core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/init/JMXServerManager.java

```diff
@@ -117,9 +120,10 @@ public void startJMXService() throws ServerException {
             JMXServiceURL url = new JMXServiceURL(jmxURL);
 
             // Security credentials are included in the env Map
-            HashMap<String, CarbonJMXAuthenticator> env =
-                new HashMap<String, CarbonJMXAuthenticator>();
+            HashMap<String, Object> env = new HashMap<String, Object>();
             env.put(JMXConnectorServer.AUTHENTICATOR, new CarbonJMXAuthenticator());
+            env.put("jmx.remote.x.access.file", ACCESS_FILE_PATH);
```

Comment:
jmx.remote.x.access.file is the environment property for JMX environment map. So it cannot be changed.